microcloud/service/lxd_config.go

```diff
@@ -0,0 +1,180 @@
+	finalUplinkCfg := api.NetworksPost{
+		NetworkPut: api.NetworkPut{
+			Config:      map[string]string{},
+			Description: "Uplink for OVN networks"},
+		Name: "UPLINK",
+		Type: "physical",
+	}
+
+	if ipv4Gateway != "" && ipv4Range != "" {
+		finalUplinkCfg.Config["ipv4.gateway"] = ipv4Gateway
+		finalUplinkCfg.Config["ipv4.range"] = ipv4Range
+	}
+
+	if ipv6Gateway != "" {
+		finalUplinkCfg.Config["ipv6.gateway"] = ipv6Gateway
+	}
+
+	ovnNetwork := api.NetworksPost{
+		NetworkPut: api.NetworkPut{Config: map[string]string{"network": "UPLINK"}, Description: "Default OVN network"},
+		Name:       "default",
+		Type:       "ovn",
+	}
+
+	return finalUplinkCfg, ovnNetwork
+}
+
+// DefaultPendingZFSStoragePool returns the default local storage configuration when
+// creating a pending pool on a specific cluster member target.
+func (s LXDService) DefaultPendingZFSStoragePool(wipe bool, path string) api.StoragePoolsPost {
+	return api.StoragePoolsPost{
+		Name:   "local",
+		Driver: "zfs",
+		StoragePoolPut: api.StoragePoolPut{
+			Config:      map[string]string{"source": path, "source.wipe": strconv.FormatBool(wipe)},
+			Description: "Local storage on ZFS",
+		},
+	}
+}
+
+// DefaultZFSStoragePool returns the default local storage configuration when
+// creating the finalized pool.
+func (s LXDService) DefaultZFSStoragePool() api.StoragePoolsPost {
+	return api.StoragePoolsPost{Name: "local", Driver: "zfs"}
+}
+
+// DefaultZFSStoragePoolJoinConfig returns the default local storage configuration when
+// joining an existing cluster.
+func (s LXDService) DefaultZFSStoragePoolJoinConfig(wipe bool, path string) []api.ClusterMemberConfigKey {
+	wipeDisk := api.ClusterMemberConfigKey{
+		Entity: "storage-pool",
+		Name:   "local",
+		Key:    "source.wipe",
+		Value:  "true",
+	}
+
+	sourceTemplate := api.ClusterMemberConfigKey{
+		Entity: "storage-pool",
+		Name:   "local",
+		Key:    "source",
+	}
+
+	sourceTemplate.Value = path
+	joinConfig := []api.ClusterMemberConfigKey{sourceTemplate}
+	if wipe {
+		joinConfig = append(joinConfig, wipeDisk)
+	}
+
+	return joinConfig
+}
+
+// DefaultPendingCephStoragePool returns the default remote storage configuration when
+// creating a pending pool on a specific cluster member target.
+func (s LXDService) DefaultPendingCephStoragePool() api.StoragePoolsPost {
+	return api.StoragePoolsPost{
+		Name:   "remote",
+		Driver: "ceph",
+		StoragePoolPut: api.StoragePoolPut{
+			Config: map[string]string{
+				"source": "lxd_remote",
+			},
+		},
+	}
+}
+
+// DefaultCephStoragePool returns the default remote storage configuration when
+// creating the finalized pool.
+func (s LXDService) DefaultCephStoragePool() api.StoragePoolsPost {
+	return api.StoragePoolsPost{
+		Name:   "remote",
+		Driver: "ceph",
+		StoragePoolPut: api.StoragePoolPut{
+			Config: map[string]string{
+				"ceph.rbd.du":       "false",
+				"ceph.rbd.features": "layering,striping,exclusive-lock,object-map,fast-diff,deep-flatten",
```

Comment:
@masnax I don't know if you know why we use these here, but it would be good to get some advice from the ceph team as to what they do and then document here as a comment why we use them.

Reply:
These are defined rather well in the ceph docs. I've linked them and provided a brief summary in the comments.